In [1]:
#import packages
import pandas as pd
import numpy as np

In [2]:
#read data
data = pd.read_csv('../Data/wiki_movie_plots_deduped.csv')
data.head()

Release Year                             Title Origin/Ethnicity  \
0          1901            Kansas Saloon Smashers         American   
1          1901     Love by the Light of the Moon         American   
2          1901           The Martyred Presidents         American   
3          1901  Terrible Teddy, the Grizzly King         American   
4          1902            Jack and the Beanstalk         American   

                             Director Cast    Genre  \
0                             Unknown  NaN  unknown   
1                             Unknown  NaN  unknown   
2                             Unknown  NaN  unknown   
3                             Unknown  NaN  unknown   
4  George S. Fleming, Edwin S. Porter  NaN  unknown   

                                           Wiki Page  \
0  https://en.wikipedia.org/wiki/Kansas_Saloon_Sm...   
1  https://en.wikipedia.org/wiki/Love_by_the_Ligh...   
2  https://en.wikipedia.org/wiki/The_Martyred_Pre...   
3  https://en.wikipedia.org/wiki/Terrible_Teddy,_...   
4  https://en.wikipedia.org/wiki/Jack_and_the_Bea...   

                                                Plot  
0  A bartender is working at a saloon, serving dr...  
1  The moon, painted with a smiling face hangs ov...  
2  The film, just over a minute long, is composed...  
3  Lasting just 61 seconds and consisting of two ...  
4  The earliest known adaptation of the classic f...

In [3]:
#get columns of data
data.columns

Index(['Release Year', 'Title', 'Origin/Ethnicity', 'Director', 'Cast',
       'Genre', 'Wiki Page', 'Plot'],
      dtype='object')

In [4]:
data = data.rename(columns= {'Origin/Ethnicity':'Origin'})

In [5]:
data.head()

Release Year                             Title    Origin  \
0          1901            Kansas Saloon Smashers  American   
1          1901     Love by the Light of the Moon  American   
2          1901           The Martyred Presidents  American   
3          1901  Terrible Teddy, the Grizzly King  American   
4          1902            Jack and the Beanstalk  American   

                             Director Cast    Genre  \
0                             Unknown  NaN  unknown   
1                             Unknown  NaN  unknown   
2                             Unknown  NaN  unknown   
3                             Unknown  NaN  unknown   
4  George S. Fleming, Edwin S. Porter  NaN  unknown   

                                           Wiki Page  \
0  https://en.wikipedia.org/wiki/Kansas_Saloon_Sm...   
1  https://en.wikipedia.org/wiki/Love_by_the_Ligh...   
2  https://en.wikipedia.org/wiki/The_Martyred_Pre...   
3  https://en.wikipedia.org/wiki/Terrible_Teddy,_...   
4  https://en.wikipedia.org/wiki/Jack_and_the_Bea...   

                                                Plot  
0  A bartender is working at a saloon, serving dr...  
1  The moon, painted with a smiling face hangs ov...  
2  The film, just over a minute long, is composed...  
3  Lasting just 61 seconds and consisting of two ...  
4  The earliest known adaptation of the classic f...

Let's first look at the "Origin" column.

In [6]:
data.Origin.unique()

array(['American', 'Australian', 'Bangladeshi', 'British', 'Canadian',
       'Chinese', 'Egyptian', 'Hong Kong', 'Filipino', 'Assamese',
       'Bengali', 'Bollywood', 'Kannada', 'Malayalam', 'Marathi',
       'Punjabi', 'Tamil', 'Telugu', 'Japanese', 'Malaysian', 'Maldivian',
       'Russian', 'South_Korean', 'Turkish'], dtype=object)

In [7]:
len(list(data.Origin.unique()))

24

Movies in this data set comes from 24 different counties/ regions.

In [8]:
data.Origin.describe() 

count        34886
unique          24
top       American
freq         17377
Name: Origin, dtype: object

There seems to be no missing values for "Origin".  

In [9]:
data.Origin.value_counts()

American        17377
British          3670
Bollywood        2931
Tamil            2599
Telugu           1311
Japanese         1188
Malayalam        1095
Hong Kong         791
Canadian          723
Australian        576
South_Korean      522
Chinese           463
Kannada           444
Bengali           306
Russian           232
Marathi           141
Filipino          128
Bangladeshi        87
Punjabi            84
Malaysian          70
Turkish            70
Egyptian           67
Assamese            9
Maldivian           2
Name: Origin, dtype: int64

Around 50% of the movies in this data set are from America.  

Let's clear the data for column 'Director' and 'Cast'.

In [10]:
# regularize the format of Director and Cast by seperate names using ',' 
data['Director'] = data['Director'].str.replace('and',',')
data['Director'] = data['Director'].str.replace('/',',')
data['Cast'] = data['Cast'].str.replace('and',',')
data['Cast'] = data['Cast'].str.replace('/',',')

In [11]:
data.Director.describe()

count       34886
unique      12593
top       Unknown
freq         1124
Name: Director, dtype: object

In [12]:
data.Director.value_counts()

Unknown                1124
Michael Curtiz           79
Hanna-Barbera            77
Lloyd Bacon              66
Jules White              63
                       ... 
Brian Clemens             1
Lester James Peries       1
Maurice Hamblin           1
Trevor Wrenn              1
Ferzan Özpetek            1
Name: Director, Length: 12593, dtype: int64

At fist it seems that there are no missing values for director, yet there are 1124 of the movies have "unknown" as director. Let's mark "Unknown" as missing. 

In [13]:
data['Director'].mask(data['Director']=='Unknown', inplace = True)

In [14]:
data.Director.value_counts()

Michael Curtiz         79
Hanna-Barbera          77
Lloyd Bacon            66
Jules White            63
John Ford              59
                       ..
Brian Clemens           1
Lester James Peries     1
Maurice Hamblin         1
Trevor Wrenn            1
Ferzan Özpetek          1
Name: Director, Length: 12592, dtype: int64

In [15]:
data.Cast.describe()

count           33464
unique          32182
top       Tom , Jerry
freq               80
Name: Cast, dtype: object

In [16]:
data.Cast.value_counts()

Tom , Jerry                                                  80
Three Stooges                                                56
The Three Stooges                                            47
Looney Tunes                                                 37
Bugs Bunny                                                   36
                                                             ..
Steve Martin, Goldie Hawn, Dana Delany                        1
Nicolas Cage, James Caan, Sarah Jessica Parker                1
Rick Moranis, Marcia Strassman                                1
Macaulay Culkin, Joe Pesci, Daniel Stern                      1
Halit Ergenç, Tuba Büyüküstün, Mehmet Günsür, Nejat İşler     1
Name: Cast, Length: 32182, dtype: int64

There are 34886 - 33464 = 1422 missing values for Cast, and there are a lot fewer replications for casts.

In [17]:
from pathlib import Path  
filepath = Path('../Data/wiki_movie_plots_deduped_c1.csv')  
filepath.parent.mkdir(parents=True, exist_ok=True)  
data.to_csv(filepath) 